crates/wasmtime/src/runtime/vm/gc.rs

```diff
@@ -96,6 +97,21 @@ impl GcStore {
         }
     }
 
+    /// Write the `source` GC reference into the uninitialized `destination`
+    /// slot, performing write barriers as necessary.
+    pub fn init_gc_ref(
+        &mut self,
+        destination: &mut MaybeUninit<Option<VMGcRef>>,
+        source: Option<&VMGcRef>,
+    ) {
+        // Initialize the destination to `None`, at which point the regular GC
+        // write barrier is safe to reuse.
+        destination.write(None);
+        let destination = unsafe { destination.assume_init_mut() };
```

Comment:
Looks like `MaybeUninit::write` returns `&mut T` so I think this can be `let destination = destination.write(None);`